cpp/src/parquet/arrow/writer.cc

```diff
@@ -480,17 +476,28 @@ class FileWriterImpl : public FileWriter {
       return Status::OK();
     };
 
+    // Max number of rows allowed in a row group.
+    const int64_t max_row_group_length = this->properties().max_row_group_length();
```

Comment:
Not sure whether we should validate that such config value is positive.
If it's set to 0, the processor would never exit the loop.